### PR TITLE
[Merged by Bors] - bump operator-rs to 0.27.1

### LINKED ISSUE #380: Bump operator-rs to 0.27.1
Update operator-rs to [0.27.1](https://github.com/stackabletech/operator-rs/blob/main/CHANGELOG.md).

- [x] operator-rs bumped
- [x] operator has no dependecy on kube or serde_yaml (use https://github.com/stackabletech/operator-rs/pull/450 instead)
- [x] Use `Fragment` wherever possible
- [x] Orphaned resources mechanism is used
- [x] Use parser for k9s `Quantity` instead of parsing memory values yourself

Product image selection will be tracked later on by https://github.com/stackabletech/issues/issues/305 but should be pretty easy compared to the changes in this Issue

### PR DESCRIPTION
**Pull request by adwk67**
# Description

operator-rs from 0.26.1 to 0.27.1
Fixes #380.

:heavy_check_mark:  https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/15/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -450,14 +456,15 @@ pub fn build_node_role_service(nifi: &NifiCluster) -> Result<Service> {
...
-            )
+            .with_recommended_labels(ObjectLabels {
+                owner: nifi,
+                app_name: APP_NAME,
+                app_version: nifi.image_version().context(NifiVersionParseFailureSnafu)?,
+                operator_name: OPERATOR_NAME,
+                controller_name: CONTROLLER_NAME,
+                role: &role_name,
+                role_group: "global",
+            })
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:467]: Do we want to have a method for this? 
```
fn build_recommended_labels<'a>(
    owner: &'a NifiCluster,
    app_version: &'a str,
    role: &'a str,
    role_group: &'a str,
) -> ObjectLabels<'a, TrinoCluster> {
    ObjectLabels {
        owner,
        app_name: APP_NAME,
        app_version,
        operator_name: OPERATOR_NAME,
        controller_name: CONTROLLER_NAME,
        role,
        role_group,
    }
}
```

File: rust/operator-binary/src/main.rs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 mod config;
 mod controller;
 
+use crate::controller::{CONTROLLER_NAME, OPERATOR_NAME};
```
Comment (maltesander) [inline rust/operator-binary/src/main.rs:4]: For me `OPERATOR_NAME` can reside even in main.rs? `CONTROLLER_NAME` should reside in the respective controller (does not matter here with only one controller, but i like the consistency :))

File: tests/templates/kuttl/upgrade/test_nifi.py [hunk 3]
```diff
@@ -37,30 +37,36 @@ def get_token(username, password, namespace):
...
+    while x < 15:
+        url = host + '/nifi-api/controller/cluster'
+        cluster = requests.get(url, headers=headers, verify=False)  # , cert='/tmp/cacert.pem')
+        if cluster.status_code != 200:
+            print("Waiting for cluster...")
+        else:
+            cluster_data = json.loads(cluster.content.decode('utf-8'))
+            nodes = cluster_data['cluster']['nodes']
+            if len(nodes) != 3:
+                print(f"Cluster should have 3 nodes at this stage, but has: {len(nodes)}")
```
Comment (maltesander) [inline tests/templates/kuttl/upgrade/test_nifi.py:57]: This checks for 3 or 2 nodes depending on the case. Does it make sense to make this another cli argument?

**Review by maltesander — COMMENTED** [review]
Some feedback, LGTM otherwise!

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (adwk67) [reply rust/operator-binary/src/controller.rs]: Done.

File: tests/templates/kuttl/upgrade/test_nifi.py (hunk 3 above)
Comment (adwk67) [reply tests/templates/kuttl/upgrade/test_nifi.py]: Yes, added.

File: rust/operator-binary/src/main.rs (hunk 2 above)
Comment (adwk67) [reply rust/operator-binary/src/main.rs]: Done.

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**